pinot-plugins/pinot-input-format/pinot-csv/src/main/java/org/apache/pinot/plugin/inputformat/csv/CSVRecordReader.java

```diff
@@ -103,6 +105,23 @@ public void init(File dataFile, @Nullable Set<String> fieldsToRead, @Nullable Re
     _recordExtractor.init(fieldsToRead, recordExtractorConfig);
   }
 
+  private void validateHeaderForDelimiter(char delimiter, String csvHeader, CSVFormat format)
+      throws IOException {
+    CSVParser parser = format.parse(RecordReaderUtils.getBufferedReader(_dataFile));
```

Comment:
we're creating a parser again in `init()` line 125. Can we avoid the duplication? Maybe move the init call to the top since it needs only the  dataFile anyway?

Reply:
CSVParser doesn't move back, so once I read a record from the datafile, it would be impossible for me to re-read it. From the docs "It is not possible to go back, once a record has been parsed from the input stream."